lib/floe/workflow/reference_path.rb

```diff
@@ -23,27 +23,15 @@ def get(context)
       end
 
       def set(context, value)
-        result = context.dup
-
-        # If the payload is '$' then merge the value into the context
-        # otherwise store the value under the path
-        #
-        # TODO: how to handle non-hash values, raise error if path=$ and value not a hash?
-        if path.empty?
-          result.merge!(value)
-        else
-          child    = result
-          keys     = path.dup
-          last_key = keys.pop
-
-          keys.each do |key|
-            child[key] = {} if child[key].nil?
-            child = child[key]
-          end
-
-          child[last_key] = value
+        return value.dup if path.empty?
+
+        child = result = context.dup
+        path[..-2].each do |key|
```

Comment:
`path[..-2]` is melting my brain...is it chomping off the last character?  If so, is it chomping off a specific character (asking cause then you can use the `chomp` method with a parameter.

Reply:
yea. I wanted to get rid of the dup, but maybe this is the same.

This is
```diff
- keys     = path.dup
- last_key = keys.pop
- keys.each do |key|
+ path[..-2].each do
  ...
  end
- child[last_key]
+ child[path.last]
```

probably same as:
```ruby
- keys     = path.dup
- last_key = keys.pop
+ *keys, last_key = path
```

Though all of these may allocate the same so this may be an unnecessary change.

Reply:
@kbrock can we tackle this in a different refactoring PR if https://github.com/ManageIQ/floe/pull/199/files#r1622463119 is the primary fix here?

Reply:
I don't know if it is any better.
Will probably forget all about this change. (as it should be)

Good call getting in the important stuff and changing as little as possible